### DISCUSSION TITLE
Unclear docs on form.watch callback variant

### DISCUSSION THREAD

**Discussion by bryanhaney-shipengine** [react-hook-form/react-hook-form#9357] [General]
It took some head scratching and source diving to figure out why the callback variant of `form.watch(() => { ... })` was getting called increasingly more times on every field change. The docs are not clear that the callback variant sets up a new subscription _per call_, and thus it belongs in a useEffect.

**Comment by bluebill1049** [comment]
Did you miss the example code below? 

https://react-hook-form.com/api/useform/watch

![Screenshot 2022-11-12 at 11 54 20 am](https://user-images.githubusercontent.com/10513364/201448683-243e71d7-d943-4f33-98ad-bdae5e6dd50d.png)

send us a PR if you have better suggestion to improve on it.
